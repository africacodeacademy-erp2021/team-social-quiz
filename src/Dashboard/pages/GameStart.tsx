import React from 'react';
import "./Category.css";
import Language from "../Images/Language.png";
import Lock from "../Images/Lock.png"
import Player from "../Images/Player.png"

function GameStart() {
    return (
        <div className="GameStart">
            <div className="PinContainer">
                <span id="PinHeader">Game PIN:</span>
            <div>
                <input className="PinInput" type="text" placeholder="583212816" />
            </div>
            <div className="LangaugeContainer">
            <img src={Language} alt="langauge" width="20px" height="20px"/>
            <span id="LangaugeText">EN</span>
            </div>
            </div>
            <div className="StartBody">
                <h4 id="StartText">quizapp</h4>
                <button id="StartButton">Start</button>
                <div className="LockContainer">
                    <img src={Lock} alt="langauge" width="20px" height="20px"/>
                </div>
                <div className="PlayerContainer">
                <img id="PlayersIcon" src={Player} alt="langauge" width="20px" height="20px"/>
                </div>
            </div>
             
        </div>
    )
}

export default GameStart;
