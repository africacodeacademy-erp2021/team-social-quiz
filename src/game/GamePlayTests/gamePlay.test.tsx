import  GameScoreFunction  from '../gameFunctions/gameScoreFunction';
import  AddScoreOnNext from '../gameFunctions/gameScoreFunction';
import { render, screen} from '@testing-library/react';
import MyAudio from '../Game_Components/audio';
import Timer from '../Game_Components/timer';
import GameStart from '../Game_Play/game_start';
import ProgressBar from '../Game_Components/progress_Bar';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { renderHook, act} from '@testing-library/react-hooks'
import { configure, shallow } from "enzyme";
configure({ adapter: new Adapter() });

test('game Audio buttons can be clicked', () => {
  render(<MyAudio/>);
  //click audio button 
  const muteBtn:any = screen.getAllByRole('button', {pressed: true});
  expect(muteBtn).toBeTruthy;
});

describe("add Score", () => {
  it("it adds score by 50", () => {    
    const { result } = renderHook(GameScoreFunction); 
     
    expect(result.current.addScoreOnNext).toBeCalled;

     })
}) 
 
//render all used components   
describe("ProgressBar", () => {
  it("renders ProgressBar correctly", () => {
    const wrapper = shallow(<ProgressBar/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Timer", () => {
  it("renders Timer correctly", () => {
    const wrapper = shallow(<Timer/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("GameStart", () => {
  it("renders GameStart correctly", () => {
    const wrapper = shallow(<GameStart/>);
    expect(wrapper).toMatchSnapshot();
  });
});


