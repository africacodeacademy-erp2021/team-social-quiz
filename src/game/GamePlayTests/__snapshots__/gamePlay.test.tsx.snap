// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameStart renders GameStart correctly 1`] = `ShallowWrapper {}`;

exports[`ProgressBar renders ProgressBar correctly 1`] = `ShallowWrapper {}`;

exports[`Timer renders Timer correctly 1`] = `ShallowWrapper {}`;
